android/composeui/src/main/java/com/stadiamaps/ferrostar/composeui/views/InstructionsView.kt

```diff
@@ -36,23 +49,60 @@ fun InstructionsView(
     distanceToNextManeuver: Double?,
     distanceFormatter: DistanceFormatter = LocalizedDistanceFormatter(),
     theme: InstructionRowTheme = DefaultInstructionRowTheme,
-    content: @Composable () -> Unit = {
-      ManeuverImage(instructions.primaryContent, tint = MaterialTheme.colorScheme.primary)
+    remainingSteps: List<RouteStep>? = null,
+    initExpanded: Boolean = false,
+    contentBuilder: @Composable (VisualInstruction) -> Unit = {
+      ManeuverImage(it.primaryContent, tint = MaterialTheme.colorScheme.primary)
     }
 ) {
-  Column(
+  var isExpanded by remember { mutableStateOf(initExpanded) }
+  val screenHeight = LocalConfiguration.current.screenHeightDp
+  val collapsedHeight = 100.dp
+
+  val targetHeight = if (isExpanded) screenHeight.dp else collapsedHeight
+  val animatedHeight by animateDpAsState(targetValue = targetHeight)
+
+  val scrollState = rememberScrollState()
+
+  Box(
       modifier =
           Modifier.fillMaxWidth()
-              .shadow(elevation = 5.dp, RoundedCornerShape(10.dp))
+              .height(animatedHeight)
               .background(theme.backgroundColor, RoundedCornerShape(10.dp))
-              .padding(8.dp)) {
-        ManeuverInstructionView(
-            text = instructions.primaryContent.text,
-            distanceFormatter = distanceFormatter,
-            distanceToNextManeuver = distanceToNextManeuver,
-            theme = theme,
-            content = content)
-        // TODO: Secondary instructions
+              .clickable { isExpanded = !isExpanded }
+              .padding(16.dp)) {
+        Column(modifier = if (isExpanded) Modifier.verticalScroll(scrollState) else Modifier) {
+          // Primary content
+          ManeuverInstructionView(
+              text = instructions.primaryContent.text,
+              distanceFormatter = distanceFormatter,
+              distanceToNextManeuver = distanceToNextManeuver,
+              theme = theme) {
+                contentBuilder(instructions)
+              }
+
+          // TODO: Secondary content
+
+          // Expanded content
+          if (isExpanded && remainingSteps != null && remainingSteps.count() > 1) {
+            Spacer(modifier = Modifier.height(8.dp))
+            HorizontalDivider(thickness = 1.dp)
+            remainingSteps.drop(1).forEach { step ->
```

Comment:
I feel like this should probably be using a [LazyColumn](https://developer.android.com/develop/ui/compose/lists#lazylistscope). Something like?

```kotlin
LazyColumn(
    verticalArrangement = Arrangement.spacedBy(16.dp)
) {
    items(remainingSteps.drop(1)) { step ->
        ManeuverInstructionView(step)
    }
}
```

Reply:
This doesn't address the horizontal rule, but it is probably marginally more efficient and lets us drop one manual spacer.